crypto/evp/keymgmt_meth.c

```diff
@@ -321,3 +335,13 @@ const OSSL_PARAM *evp_keymgmt_export_types(const EVP_KEYMGMT *keymgmt,
         return NULL;
     return keymgmt->export_types(selection);
 }
+
+int evp_keymgmt_copy(const EVP_KEYMGMT *keymgmt,
+                     void *keydata_to, const void *keydata_from,
+                     int selection)
+{
+    /* We assume no copy if the implementation doesn't have a function */
+    if (keymgmt->copy == NULL)
+        return 0;
```

Comment:
Why do we return 0 instead of -2 here?

Reply:
-2 isn't a global value.  It isn't even consistent throughout all of EVP, there are some parts that return -1.  I did add a TODO in `keymgmt_lib.c`, in the middle of `evp_keymgmt_util_export_to_provider`, addressing exactly this question.  Looks like this:

``` C
    /* If the "origin" |keymgmt| doesn't support exporting, give up */
    /*
     * TODO(3.0) consider an evp_keymgmt_export() return value that indicates
     * that the method is unsupported.
     */
    if (pk->keymgmt->export == NULL)
        return NULL;
```

So *for the moment*, we're checking the function pointer before calling the `evp_keymgmt` wrapper function.  One day, we may get sick and tired of it and reconsider this question, but considering it's all internal functions, it doesn't have to be addressed today.